primitives/io/src/lib.rs

```diff
@@ -166,11 +191,54 @@ pub trait Storage {
 	/// Use this function to distribute the deletion of a single child trie across multiple
 	/// blocks.
 	#[version(2)]
+	fn clear_prefix(&mut self, prefix: &[u8], limit: Option<u32>) -> KillStorageResultPrev {
+		let clear_prefix_result = Externalities::clear_prefix(*self, prefix, limit);
+		match clear_prefix_result.are_keys_remaining {
+			true => KillStorageResultPrev::AllRemoved(clear_prefix_result.num_keys_from_backend),
+			false =>
+				KillStorageResultPrev::SomeRemaining(clear_prefix_result.num_keys_from_backend),
+		}
+	}
+
+	/// Clear the storage of each key-value pair where the key starts with the given `prefix`.
+	///
+	/// # Limit
+	///
+	/// Deletes all keys from the overlay and up to `limit` keys from the backend if
+	/// it is set to `Some`. No limit is applied when `limit` is set to `None`.
+	///
+	/// The limit can be used to partially delete a prefix storage in case it is too large
+	/// to delete in one go (block).
+	///
+	/// It returns a boolean false iff some keys are remaining in
```

Comment:
which boolean ?

Reply:
I was trying to ride on the comments from v2, because that's what i saw there

Reply:
yes it is probably from an artefact from a previous implementation, it probably means AllRemoved/SomeRemaining variant